src/OpenTelemetry.AutoInstrumentation.Native/integration_loader.cpp

```diff
@@ -98,8 +108,45 @@ void LoadIntegrationsFromStream(std::istream& stream, std::vector<IntegrationMet
 
 namespace
 {
+    bool InstrumentationEnabled(const WSTRING name, const std::vector<WSTRING>& enabledIntegrationNames, const std::vector<WSTRING>& disabledIntegrationNames)
+    {
+        // LoggingBuilder has to be always enabled/
+        // Technically it is  non-instrumentation but
+        // it is using the same functionality.
+        // See https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1310.
+        if (name == WStr("LoggingBuilder"))
+        {
+            return true;
+        }
 
-    void IntegrationFromJson(const json::value_type& src, std::vector<IntegrationMethod>& integrationMethods, const std::vector<WSTRING>& disabledIntegrationNames)
+        // check if the integration is disabled
+        for (const WSTRING& disabledName : disabledIntegrationNames)
+        {
+          if (name == disabledName) 
+          {
+              return false;
+          }
+        }
+
+        if (enabledIntegrationNames.empty())
+        {
+            return true;
+        }
+        // check if the integration is enabled
```

Comment:
Should this be other way, first check enabled Integration and then check for disabled ones? 
In the managed layer we check enabled instrumentations first - https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/d6c9ffad4de5686ec856a3c6ca1e243928e9f44f/src/OpenTelemetry.AutoInstrumentation/Util/ConfigurationSourceExtensions.cs#L26

Reply:
I think that the output of both "algorithms" is the same 🤔

Reply:
May not be same. If I have same entries in both enabled and disabled instrumentation, it will give different results. If we have entry in both places we should disable the instrumentation.

Reply:
Agree and this is what this code does as well 😉 If `return false` and even does not check if it is enabled.

Reply:
Thanks for correcting me. I completely missed we are just checking based on single entry `name`.